Listview2

jonte450 opened the pull request:
hi

jonte450 committed:
Added a Listview

jonte450 committed:
Connect to the database in flutter

0x6273 commented on spring-backend/src/main/java/com/perfect8/blog/service/PostService.java:
```diff
@@ -20,7 +21,13 @@ public PostService(PostRepository postRepository) {
 
-    /// Get a post by its unique URL slug.
+
+public List<PostEntity> getAllPosts() {
+    return (List<PostEntity>) postRepository.findAll();
+}
+
+
+/// Get a post by its unique URL slug.
```
Det blev fel på indenteringen här tror jag.

```suggestion
    public List<PostEntity> getAllPosts() {
        return (List<PostEntity>) postRepository.findAll();
    }


    /// Get a post by its unique URL slug.
```

0x6273 commented on flutter-frontend/perfect8_front_end/lib/post.dart:
```diff
@@ -0,0 +1,42 @@
+      throw Exception('Failed to load posts');
+    }
+  }
+
+  Future<Post> fetchPostBySlug(String slug) async {
+    final response = await http.get(Uri.parse('$baseUrl/$slug'));
+    if (response.statusCode == 200) {
+      return Post.fromJson(jsonDecode(response.body));
+    } else {
+      throw Exception('Post not found');
+    }
+  }
```
Den här används inte? Vi har redan en `fetchPost` i `api/post.dart` som gör samma sak.

```suggestion
```

0x6273 committed:
Fixes

- flutter: Move `post.dart` to `api/post_preview.dart` and rename
  classes.
- spring: Comment out debugging.
- spring: Rename SlugDTO `dateTime` to more descriptive `publishedAt`
- spring: Rename SlugDTO to PostPreviewDto
- spring: Rename SlugToDTO to PostPreviewDtoMapper
- spring: Add more comments
- spring: Whitespace fixes

0x6273 committed:
More fixes
